pkg/controller/mysql/grant/reconciler.go

```diff
@@ -0,0 +1,324 @@
+			ResourceUpToDate: false,
+		}, nil
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: true,
+	}, nil
+}
+
+func comparePrivileges(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	sort.Strings(a)
+	sort.Strings(b)
+
+	for i := range a {
+		// Special case because ALL is an alias for "ALL PRIVILEGES"
+		strA := strings.ReplaceAll(a[i], allPrivileges, "ALL")
+		strB := strings.ReplaceAll(b[i], allPrivileges, "ALL")
+		if strA != strB {
+			return false
+		}
+	}
+	return true
+}
+
+func parseGrant(grant, dbname string) (privileges []string) {
+	matches := grantRegex.FindStringSubmatch(grant)
+	if len(matches) == 3 && matches[2] == dbname {
+		return strings.Split(matches[1], ", ")
+	}
+	return nil
+}
+
+func (c *external) getPrivileges(ctx context.Context, username, dbname string) ([]string, *managed.ExternalObservation, error) {
+	username, host := mysql.SplitUserHost(username)
+	query := fmt.Sprintf("SHOW GRANTS FOR %s@%s", mysql.QuoteValue(username), mysql.QuoteValue(host))
+	rows, err := c.db.Query(ctx, xsql.Query{String: query})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, errCurrentGrant)
+	}
+	defer rows.Close() //nolint:errcheck
+
+	var privileges []string
+	for rows.Next() {
+		var grant string
+		if err := rows.Scan(&grant); err != nil {
+			return nil, nil, errors.Wrap(err, errCurrentGrant)
+		}
+		p := parseGrant(grant, dbname)
+		if p != nil && privileges != nil {
+			// Found more than one grant for this user on this DB
+			return nil, &managed.ExternalObservation{
+				ResourceExists:   true,
+				ResourceUpToDate: false,
+			}, nil
+		}
+		privileges = p
+	}
+	if err := rows.Err(); err != nil {
+		var myErr *mysqldriver.MySQLError
+		if errors.As(err, &myErr) {
+			if myErr.Number == errCodeNoSuchGrant {
+				// The user doesn't (yet) exist and therefore no grants either
+				return nil, &managed.ExternalObservation{ResourceExists: false}, nil
+			}
+		}
+		return nil, nil, errors.Wrap(err, errCurrentGrant)
+	}
+	if privileges == nil {
+		return nil, &managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	return privileges, nil, nil
+}
+
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Grant)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotGrant)
+	}
+
+	username := *cr.Spec.ForProvider.User
+	dbname := *cr.Spec.ForProvider.Database
+
+	privileges := strings.Join(cr.Spec.ForProvider.Privileges, ", ")
+
+	query := createGrantQuery(privileges, dbname, username)
+	if err := c.db.Exec(ctx, xsql.Query{String: query}); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateGrant)
+	}
+	if err := c.db.Exec(ctx, xsql.Query{
+		String: "FLUSH PRIVILEGES",
+	}); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errFlushPriv)
+	}
+	return managed.ExternalCreation{}, nil
```

Comment:
Nit: `errors.Wrap` will return `nil` if `err` is `nil`, so you can just write:

```go
err := c.db....
return managed.ExternalCreation{}, errors.Wrap(err, errFlushPriv)
```

Ditto for `Update` and `Delete`.